### ISSUE TITLE
An error occurred while handling message: GET_RUN_ARGUMENTS

### ISSUE THREAD

**Issue by git119** [Microsoft/vscode-cordova#368]
- Extension Name: cordova-tools
- Extension Version: 1.3.3
- OS Version: Windows_NT x64 10.0.16299
- VSCode version: 1.20.0

 We have written the needed data into your clipboard. Please paste:
  {
            "name": "Serve to the browser (ionic serve)",
            "type": "cordova",
            "request": "launch",
            "platform": "serve",
            "devServerAddress": "localhost",
            "sourceMaps": true,
            "ionicLiveReload": true,
            "cwd": "${workspaceRoot}"
        } 

[cordova-tools] Error processing "launch": Error: An error occurred while handling message: GET_RUN_ARGUMENTS

**Comment by maxi-jp** [comment]
I was just going to comment this to. It happens to me with almost all configurations (I've tested with these): 

`
        {
            "name": "Simulate android|ios in browser",
            "type": "cordova",
            "request": "launch",
            "platform": "android|ios",
            "target": "chrome",
            "simulatePort": 8000,
            "livereload": true,
            "cwd": "${workspaceRoot}",
            "sourceMaps": true
        },
        {
            "name": "Run Android on emulator",
            "type": "cordova",
            "request": "launch",
            "platform": "android",
            "target": "emulator",
            "port": 9222,
            "sourceMaps": true,
            "cwd": "${workspaceRoot}",
            "ionicLiveReload": false
        },
        {
            "name": "Simulate Android in browser",
            "type": "cordova",
            "request": "launch",
            "platform": "android",
            "target": "chrome",
            "simulatePort": 8000,
            "livereload": true,
            "sourceMaps": true,
            "cwd": "${workspaceRoot}"
        },
        {
            "name": "Run Browser",
            "type": "cordova",
            "request": "launch",
            "platform": "browser",
            "target": "chrome",
            "simulatePort": 8000,
            "livereload": true,
            "sourceMaps": true,
            "cwd": "${workspaceRoot}"
        }
`

**Comment by tnovau** [comment]
Same error here...

cordova (Cordova CLI) : 8.0.0
OS   : Windows 10
Version of Vscode: 1.20.0

**Comment by lucamorelli** [comment]
same problem here

**Comment by ROSteh** [comment]
see this problem...
terminal go command: cordova run
then "Attach to running" for android: 

> {
>             "name": "Attach to running android on device",
>             "type": "cordova",
>             "request": "attach",
>             "platform": "android",
>             "target": "device",
>             "port": 9222,
>             "sourceMaps": true,
>             "cwd": "${workspaceRoot}"
>         },

**Comment by Wilco2998** [comment]
same problem here and can't attach to running browser

**Comment by grozdanovgg** [comment]
I have the same problem since yesterday

**Comment by sergey-akhalkov** [comment]
Hey guys, thanks for reaching us, we are investigating it and we'll let you know if we have any updates.

**Comment by coloz** [comment]
same problem after  update  vscode-1.20.0

**Comment by RobertPinson** [comment]
same problem after update vscode-1.20.0

**Comment by satya5561** [comment]
see this problem...

**Comment by darshitmaniar** [comment]
same problem after update vscode-1.20.0

**Comment by max-mironov** [comment]
Hey @git119, @maxi-jp, @tnovau, @lucamorelli and all others, thanks for reporting and for your patience. 
Could you please try this PR https://github.com/Microsoft/vscode-cordova/pull/370 and let us know if it helps. If yes we would make new release in the nearest days.

**Comment by maxi-jp** [comment]
It wasn't working for me:
`[cordova-tools] Error processing "launch": TypeError: Cannot read property 'catch' of undefined`

I have deleted the launch.json file for regenerating it again but now in the "Configure or Fix launch.json" button in the Debug tab it doesn't appers the Cordova option D:

**Comment by annakocheshkova** [comment]
@maxi-jp I have just tested the fix on Windows: launched extension -> Chose Debug -> Run android on Emulator. Could you please share with us your repro steps and configuration you use? The fix doesn't seem to be breaking anything connected with `catch` block, so it looks like another issue, not the one connected with the fix.

**Comment by maxi-jp** [comment]
Sorry, I wasn't editing the extensionMessaging.js properly, now something has changed but still doesn't work. Selecting "Simulate Android in browser" it says:
```
Launching for android (This may take a while)...
An error occurred while attaching to the debugger. Error starting the simulation
Error starting the simulation
```

**Comment by annakocheshkova** [comment]
@maxi-jp got it, looks like the issue persists with the configuration you specified, although not quite sure if it's connected with the one above. Will continue to investigate on that.

**Comment by lucamorelli** [comment]
looks like it's working for me with a real device

**Comment by max-mironov** [comment]
@maxi-jp seems that your issue is similar to https://github.com/Microsoft/vscode-cordova/issues/361#issuecomment-363777711
Looks like not related to the current one, so I think would be better to create a separate one (or probably better to reopen previously created)

**Comment by max-mironov** [comment]
New version has been released: https://github.com/Microsoft/vscode-cordova/releases/tag/1.3.4
Please let us know if you see any issues.

**Comment by maxi-jp** [comment]
Thanks @max-mironov 
I don't have an assets folder either in platforms/android, but cordova is able to create a working apk via `cordova build android`.
I think I have to investigate further to see whats really going on in my PC with VSC, cordova and this plugin. If I find any new clue I'll let you know.

**Comment by jprinsloo** [comment]
The new release works for me...it does however have an issue with the platform pipe operator _android|ios_. I have to manually specify android or ios.

**Comment by max-mironov** [comment]
@maxi-jp thanks, please reach us if you find any clues.
@jprinsloo got it, I believe I saw this behavior previously also. Do you think this could be related to 1.20 version upgrade?

**Comment by satya5561** [comment]
@git119  I have uninstall Cordova Tools Extension and reinstall it. issue resolved .

**Comment by git119** [comment]
The new release works for me. thanks

**Comment by strogof** [comment]
The new release does not works for me.

Luigi

![image](https://user-images.githubusercontent.com/12082024/36218384-7565cc12-11b4-11e8-9bca-e94e82b64543.png)

**Comment by sergey-akhalkov** [comment]
Hi @strogof, thanks for reaching us, could you please create new issue and provide us as much info as possible about your environment?

**Comment by strogof** [comment]
Hi, Sergey,
sorry for boring you..
it was a false problem, after a restart of windows ,the issue disappeared
I tried to edit my post, but i can't find it
you can close

thanks and have a nice day

Luigi


2018-02-15 10:08 GMT+01:00 Sergey Akhalkov <notifications@github.com>:

**Comment by sergey-akhalkov** [comment]
Hi @strogof, great! Nice to hear it works for you, have a good day too.

**Comment by paqogomez** [comment]
I also had this problem, but simply restarting VS Code and the issue went away.  (I had had VS Code up for a couple of days)

**Comment by rlam3** [comment]
Still getting this error. Was wondering if there was a workaround. 

```json
 {
      "name": "Simulate iOS in browser",
      "type": "cordova",
      "request": "launch",
      "platform": "ios",
      "target": "chrome",
      "simulatePort": 8000,
      "livereload": true,
      "sourceMaps": true,
      "cwd": "${workspaceRoot}"
    },
```

```console
EACCES: permission denied, open 'local-storage.json'
```

**Comment by pammacdotnet** [comment]
Any news...? Experimenting the same issue…

**Comment by pammacdotnet** [comment]
Seems to work with "Simulate iOS in Browser", but not with "Simulate Android in Browser".

**Comment by annakocheshkova** [comment]
@rlam3, seems like a whole different issue. If the issue persists for you, go ahead and feel free to open a new one.
@pammacdotnet, please tell us a little bit more which one of the issues still has place for you?

**Comment by annakocheshkova** [comment]
@rlam3 I have investigated on the issue a little bit more and it turned out that the issue itself exists, but not related to our extension. In fact, it is related to `cordova-simulate`. I have create a [PR ](https://github.com/Microsoft/cordova-simulate/pull/260) that fixes it, so you can either wait until it's merged and re-install the plugin, or find a workaround.
I have 2 suggestions for you:
* Try running simulate once again after the issue occurs - in my case the second time everything worked fine.
* Go to `~/.vscode/extensions/vsmobile.cordova-tools-1.3.5/node_modules/cordova-simulate/src/server/utils/jsUtils.js` (you may need to additionally verify path to `.vscode` folder and name of the extension folder).  Find `getAppDataPath()` method there and change the following lines
```
case 'darwin':
            appDataPath = process.env.HOMEPATH;
```
to the following lines:
```
case 'darwin':
            appDataPath = process.env.HOMEPATH || process.env.HOME;
```

**Comment by max-mironov** [comment]
Closing this item for now as the upper issue is investigated and not related to the original one. 
However regarding the last issue - we will release new version as soon as the new version of `cordova-simulate` will be released. 
Please let us know if you see any issues or have any questions.

**Comment by jmesa-sistel** [comment]
'ionic cordova platform add android ' command is not creating www folder, I think this is the problem.

**Comment by eddskt** [comment]
Hello, I'm using vs code 1.22 and I'm having the same problem on console.
![image](https://user-images.githubusercontent.com/6931017/39459389-7e2c8cca-4cd1-11e8-899a-2355926a656d.png)

**Comment by MathewLC** [comment]
I having the same issue but in windows OS. Do you have any answer????

**Comment by eddskt** [comment]
I solved the problem just by doing the following:

**1** - cordova platform rm android
**2** - Removing a platform I went into the Visual Studio Community and did a compilation in debug mode in the browser.
**3** - Without platform, he asked to add. I clicked Yes

I do not know if it was incompatibility on the platform or in the dependencies yet.
But adding the platform by visual studio, he corrected the error.

I think it's some incompatibility in dependencies or plugins.
Well he warned me when adding the platform that some plugins might not work.
I just do not remember which
